File: lista java/at15.java

public class SaldoInsuficienteException extends Exception {
    public SaldoInsuficienteException(double saldo, double valorSaque) {
        super("Saldo insuficiente! Saldo atual: " + saldo + ", Valor do saque: " + valorSaque);
    }
}


public class ContaBancaria {
    private String titular;
    private double saldo;

    public ContaBancaria(String titular, double saldo) {
        this.titular = titular;
        this.saldo = saldo;
    }

    public void sacar(double valor) throws SaldoInsuficienteException {
        if (valor > saldo) {
            throw new SaldoInsuficienteException(saldo, valor);
        }
        saldo -= valor;
    }

    public void depositar(double valor) {
        saldo += valor;
    }

    public double getSaldo() {
        return saldo;
    }

    public static void main(String[] args) {
        ContaBancaria conta = new ContaBancaria("João", 500);

        try {
            conta.sacar(600);  
        } catch (SaldoInsuficienteException e) {
            System.out.println(e.getMessage());
        }
    }
}

File: lista python/at15.py
class SaldoInsuficienteException(Exception):
    def __init__(self, saldo, valor_saque):
        self.saldo = saldo
        self.valor_saque = valor_saque
        super().__init__(f"Saldo insuficiente! Saldo atual: {self.saldo}, Valor do saque: {self.valor_saque}")

class ContaBancaria:
    def __init__(self, titular, saldo=0):
        self.titular = titular
        self.saldo = saldo

    def sacar(self, valor):
        if valor > self.saldo:
            raise SaldoInsuficienteException(self.saldo, valor)
        self.saldo -= valor
        return valor

    def depositar(self, valor):
        self.saldo += valor


conta = ContaBancaria("Jose", 5000)

try:
    conta.sacar(5100)  
except SaldoInsuficienteException as e:
    print(e)
